lib/kolekti_analizo/collector.rb

```diff
@@ -61,9 +61,17 @@ def run(code_directory)
       end
 
       def metric_list
-        list = `analizo metrics --list`
-        raise Kolekti::CollectorError.new("Analizo failed with exit status: #{$?.exitstatus}") unless $?.success?
-        list
+        readio, writeio = IO.pipe
+
+        success = system('analizo metrics --list', out: writeio)
+        writeio.close
+
+        if success
+          readio.read
+        else
+          raise Kolekti::CollectorError.new("Analizo failed")
```

Comment:
This depends more on analizo correct outputing to stderr (please notice we have not redirected it). Getting the exit status was not providing any extra information and was just breaking mock behaviour.